test/test_ad.jl

```diff
@@ -115,6 +115,14 @@ _J1 = forwarddiff_color_jacobian(oopf, x, colorvec = repeat(1:3,10), sparsity =
 @test _J1 ≈ J
 @test fcalls == 1
 
+#oop with in-place Jacobian
+fcalls = 0
+_oop_jacout = sparse(1.01 .* J) # want to be nonzero to check that the pre-allocated matrix is overwritten properly
+forwarddiff_color_jacobian(_oop_jacout, oopf, x; colorvec = repeat(1:3,10), sparsity = _J, jac_prototype = _J)
```

Comment:
add a test on this call which is sparse but not SparseMatrixCSC, like BandedMatrix.